scripts/plugins/extauth-hook-AD.py

```diff
@@ -233,6 +223,55 @@ def _install(self):
                 os.remove(self._file_path)
 
 
+class UsersList(DynamicPam):
+    #pylint: disable=too-few-public-methods
+    """Class manage users which permit pool admin ssh"""
+
+    def __init__(self, session, arg, ad_enabled=True):
+        super(UsersList, self).__init__(HCP_USERS, session, arg, ad_enabled)
+
+    def _match_subject(self, subject_rec):
+        return subject_rec["other_config"]["subject-is-group"] != "true"
+
+    def _add_subject(self, subject_rec):
+        try:
+            sid = subject_rec['subject_identifier']
+            name = subject_rec["other_config"]["subject-name"]
+            formatted_name = self._format_item(name)
+            logger.debug("Permit user %s with sid %s", formatted_name, sid)
+            self._lines.append(formatted_name)
+            # If ssh key is permittd in authorized_keys,
+            # The original name is compared, add UPN and original name
+            if self._backend == ADBackend.BD_PBIS and name != formatted_name:
+                self._lines.append(name)
+            self._lines.append(subject_rec["other_config"]["subject-upn"])
```

Comment:
When this fails logs will show "Failed to add user", I think that's bound to cause confusion when triaging issues around this code, I'd rather have a different logline to be shown, if at all.

This could be done with:
```python
            if "subject-upn" in subject_rec["other_config"]:
                self._lines.append(subject_rec["other_config"]["subject-upn"])
            else:
                logger.info("UPN not found for user %s", subject_rec)
```

Reply:
updated, could you please help to check?